app/helpers/functions.js

```diff
@@ -31,13 +34,114 @@ module.exports = {
         }
     },
 
-
+    sanitizeString: function(source){
+        var ret = sanitizer.escape(source);
+        return ret;
+    },
 
     getClientIp: function(){
         return ip.address();
     },
 
+    signHmacSha512(key, str) {
+        let hmac = crypto.createHmac("sha512", key);
+        let signed = hmac.update(Buffer.from(str, 'utf-8')).digest("hex");
+        return signed
+    },
+
+    getKeySHA512(){
+        let key = "kjsdfjhsdfkljh.,mnsdf15213432t13432516s8d64fs sdlkjfrlk4n5t,nskndfs*#@$%#@$%907lkkl.jalksjdlkajsdlkjaslkdj_+3234234"; //SHA512 Key Standart
+        return key;
+    },
+
+    encrypt(text) {
+        let cipher = crypto.createCipheriv(algorithm, Buffer.from(key), iv);
+        let encrypted = cipher.update(text.toString());
+        encrypted = Buffer.concat([encrypted, cipher.final()]);
+        return encrypted.toString('hex');
+    },
+
+    decrypt(text) {
+        // let iv = Buffer.from(text.iv, 'hex');
+        let encryptedText = Buffer.from(text, 'hex');
+        let decipher = crypto.createDecipheriv(algorithm, Buffer.from(key), iv);
+        let decrypted = decipher.update(encryptedText);
+        decrypted = Buffer.concat([decrypted, decipher.final()]);
+        return decrypted.toString();
+    },
+
+    encryptPassword(known_string){ //Author Prayogi
+        let key = this.getKeySHA512(); //SHA512 Key Standart
+        let hmac = crypto.createHmac("sha512", key);
+        let signed = hmac.update(Buffer.from(known_string, 'utf-8')).digest("hex");
+
+        console.log('known_string', known_string);
+        console.log('signed', signed);
+
+        return signed
+    },
+
+    validatePassword(known_string, encrypted_string){ //Author Prayogi
+        let key = this.getKeySHA512(); //SHA512 Key Standart
+        let hmac = crypto.createHmac("sha512", key);
+        let signed = hmac.update(Buffer.from(known_string, 'utf-8')).digest("hex");
+
+        console.log('known_string', known_string);
+        console.log('signed', signed);
+        console.log('encrypted_string', encrypted_string);
+
+        try{
+            if(hashEquals(signed, encrypted_string)){
+                return true;
+            }else{
+                return false;
+            }
+        }catch(err){
+            return false;
+        }
+    },
+
+    encryptToken(text) {
+        const AES_METHOD = 'aes-256-cbc';
+        const IV_LENGTH = 16; // For AES, this is always 16, checked with php
+
+        const password = 'abcdefghijklmnopqrstuvwxyz123456$'; // Must be 256 bytes (32 characters)

```

Comment:
If you pass constant then it can't be changed later in code, anything that you want to change later on in code means you expect the value of particular variable will change at any part of your code you must pass let variable_name = "value".